checkST2110.js

```diff
@@ -971,210 +1164,481 @@ const test_30_62_4 = (sdp, params) => {
+    }
     return [];
   }
-  let [ mtParams, errors ] = extractMTParams(sdp, params);
-  for ( let stream of mtParams ) {
-    if ( typeof stream.CMAX !== 'undefined' ) {
+  let [mtParams, errors] = extractMTParams(sdp, params);
+  for (let stream of mtParams) {
+    if (typeof stream.CMAX !== 'undefined') {
       let cmax = +stream.CMAX;
       if (isNaN(cmax)) {
-        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter is not a number, as per SMPTE ST 2110-21 Section 8.2.`));
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter is not a number, as per ST 2110-21 Section 8.2.`));
         continue;
       }
       if (Number.isInteger(cmax) === false) {
-        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter in not an integer, as per SMPTE ST 2110-21 Section 8.2.`));
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter in not an integer, as per ST 2110-21 Section 8.2.`));
       }
       if (cmax < 1) {
-        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter makes no sense unless it is a positive value, as per SMPTE ST 2110-21 Section 8.2.`));
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter makes no sense unless it is a positive value, as per ST 2110-21 Section 8.2.`));
+      }
+    }
+    else {
+      if (params.verbose) {
+        console.log('Test Skipped: ST 2110-21 Section 8.2 Test 2 - CMAX Not Present.');
       }
     }
   }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-21 Section 8.2 Test 2 - When traffic shaping and CMAX parameter specified, it is an acceptable value.');
+  }
   return errors;
 };
 
+const typesPermitted22 = ['2110TPN', '2110TPNL', '2110TPW'];
+
+// ST 2110-22 Section 5.3 Test 1 - Traffic Shaping and Delivery Timing - TP parameter is an acceptable value
+const test_22_53_1 = (sdp, params) => {
+  if (params.shaping === false || params.audioOnly === true) {
+    if (params.verbose) {
+      console.log('Test Skipped: ST 2110-22 Section 5.3 Test 1 - TP parameter is an acceptable value. Use --shaping to test.');
+    }
+    return [];
+  }
+  let [mtParams, errors] = extractMTParams(sdp, params);
+  for (let stream of mtParams) {
+    if (typeof stream.TP !== 'undefined') {
+      if (typesPermitted22.indexOf(stream.TP) < 0) {
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, format parameter 'TP' is not one of '2110TPN', '2110TPW', '2110TPNL', as per ST 2110-22 Section 5.3`));
+      }
+    }
+  }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-22 Section 5.3 Test 1 - TP parameter is an acceptable value.');
+  }
+  return errors;
+};
+
```

Comment:
Hmm, `CMAX` is also a permitted parameter in ST 2110-22... so should there be an equivalent/identical test to `test_21_82_2`? (I'm not sure about `TROFF` with ST 2110-22, but maybe that like `test_21_82_1` too...)